### DISCUSSION TITLE
Filter form

### DISCUSSION THREAD

**Discussion by tpaksu** [livewire/livewire#1848] [1. Help] [answered]
Hello, 

I'm out of ideas and I wanted to ask, I had a full page component with filter form and product list on the same component, but on every input change, it was re-rendering all the component, which takes time, and doesn't look pretty. Then I decided to separate the filter form and the list, but then I started making double requests when I change an input (first to update the bound member, and second to send the parameters to the list component). Is there any other practice? What would be the optimal solution for this case?

Thanks.

**Comment by iksaku** [comment, marked as answer]
The most common approach is, as you first mentioned, to update your list after every change to your filter values. Another way could be to use `wire:model.defer` and have an action called `applyFilters`, which, when called, will apply all your filters.

There's no "best way", it will always depend on what you're trying to achieve and how you would **like** it to work.

**Reply by tpaksu** [reply]
Yep, I ended with having deferred inputs on the filter form (except categories and brands), and for other filters, added a "apply filters" button. The button sends an event with the filter values to the listing component. This seemed to be the less resource using way.
